src/dbnode/persist/fs/types.go

```diff
@@ -325,6 +347,63 @@ type IndexFileSetReader interface {
 	Validate() error
 }
 
+// FileStatus describes the current file status.
```

Comment:
would these types be better placed in `x/os` or some such?

Reply:
(FileStatus/File/Registry/..)

Reply:
Yeah, having different `File` interfaces for mmap, digest, and persist/fs is likely to cause confusion I think and we'll probably have to use some conversion logic between them at some point.